src/FsCodec/FsCodec.fsproj

```diff
@@ -21,6 +22,13 @@
     <PackageReference Include="FSharp.Core" Version="3.1.2.5" Condition=" '$(TargetFramework)' == 'net461' " />
     <PackageReference Include="FSharp.Core" Version="4.3.4" Condition=" '$(TargetFramework)' == 'netstandard2.0' " />
     <PackageReference Include="FSharp.UMX" Version="1.0.0" />
+    <PackageReference Include="TypeShape" Version="8.0.0" />
```

Comment:
I'm slightly reticent to add this dependency at this level.
My thinking is that the `FsCodec` package to date has largely only supplied contracts, and does not force many dependencies.
The scenario I had in my head is if someone is using an external library to provide the Codecs (and/or you're using the `Create` overloads that take an `encode` and `tryDecode` pair).
I'm not sure this matters.

This also slightly overlaps with my stance on using `InternalsVisible` and making things fully DRY by stuffing things into the core. My preferred technique is to have a source file somewhere in the tree and then add it as a reference to multiple projects. This allows one to e.g. keep FsCodec at `2.0.0` while still allowing one to update `FsCodec.SystemTextJson` to `2.4.0` without necessitating the same change for other modules in a subsystem that still use `FsCodec.NewtonsoftJson`.

I think this pattern should be followable here?